assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -4445,6 +4488,59 @@ protected String build_instructor_grade_assignment_context(VelocityPortlet portl
 
 	} // build_instructor_grade_assignment_context
 
+	protected void build_show_students_additional_information_context(Context context, SessionState state){
+
+		List<SubmitterSubmission> returnResources = new ArrayList();
+		String aRef = (String) state.getAttribute(EXPORT_ASSIGNMENT_REF);
+		String contextString = (String) state.getAttribute(STATE_CONTEXT_STRING);
+		Site sst = null;
+		try {
+			sst = SiteService.getSite(contextString);
+			context.put("isAdditionalNotesEnabled", candidateDetailProvider.isAdditionalNotesEnabled(sst));
+			context.put("candidateDetailProvider", candidateDetailProvider);
+			context.put("site", sst);
+		
+			Map<User, AssignmentSubmission> submitters = AssignmentService.getSubmitterMap(Boolean.FALSE.toString(), "all", null, aRef, contextString);
+			for (User u : submitters.keySet())
+			{
+				if(!candidateDetailProvider.getAdditionalNotes(u, sst).isPresent()){
+					M_log.debug("Skipping user with no additional notes " + u.getEid());
+					continue;
+				}
+	
+				AssignmentSubmission sub = submitters.get(u);
+				SubmitterSubmission us = new SubmitterSubmission(u, sub);
+				returnResources.add(us);
+			}
+	
+			String ascending = "true";
+			String sort = "sorted_grade_submission_by_lastname";
+			String anon = (String) state.getAttribute(NEW_ASSIGNMENT_CHECK_ANONYMOUS_GRADING);
+			context.put("value_CheckAnonymousGrading", anon);
+			AssignmentComparator ac = new AssignmentComparator(state, sort, ascending);
+			if(Boolean.valueOf(anon)){
+				ac.setAnon(Boolean.valueOf(anon));
+			}
+			try
+			{
+				Collections.sort(returnResources, ac);
+			}
+			catch (Exception e)
+			{
+				// log exception during sorting for helping debugging
+				M_log.warn(this + ":build_show_students_additional_information_context sort=" + sort + " ascending=" + ascending + " " + e.getStackTrace());
```

Comment:
Should always pass the exceptions as an additional argument rather than calling `e.getStacktrace()`